### PR TITLE
Update query interface to use filterable / multi-select inputs for device and animal

### PR DESCRIPTION
**Pull request by arthurmcgregor**
Closed #88 and #90 and integrated into this PR.
References #81 

Remaining issues to solve:
- [x] X button not working on Firefox
- [x] Add scrollable menu to animal list
- [x] Check nothing else broke when I re-wrote the code to remove the conditions variable

The X button not working is a firefox thing - it works fine for me in Chrome. I can see the event listener attached to the X' span element but it doesn't seem to fire the event listener. Any ideas @CameronRP ?

remove device issue

### CODE REVIEW

**Comment by arthurmcgregor** [comment]
I've added the scrollable menu to the animal list (to match the device list), but it now means that you can't see all of the options - in particular interesting and false-positive are at the bottom. Perhaps let me know the most common animals/tags and I'll put them at the top of the list for easy clickability.

**Comment by mjs** [comment]
I can't imagine we'll want to query false-positives directly that often so that can stay down the bottom. Similarly non-mammals like birds and spiders can stay down (I don't know why we even bother having a spider tag!).

Perhaps start the list in this order: "interesting", possum, rat, stoat, ferret, weasel, hedgehog, cat
After that it doesn't matter so much aside from the low priority ones already mentioned being right at the bottom.

**Comment by mjs** [comment]
Could this be the event issue: https://stackoverflow.com/questions/26402033/missing-click-event-for-span-inside-button-element-on-firefox#26403202

**Comment by arthurmcgregor** [comment]
Great thanks - that fixed the Firefox bug :-) I've also done a few other minor fixes. 

I now think this is ready to go, so long as nothing else has slipped through in my code re-write. Will close #81 in my opinion 👍

**Comment by mjs** [comment]
This is looking great! I can confirm that the previous issues I mentioned have been fixed.

**Comment by mjs** [comment]
One issue is that the dropdown appears to be have some residual state when you make the next selection. For example:

image

Here I've just selected livingsprings02 and when I go to select the next device the list is still filtered based on what I had typed before even though the text box is empty. You then have to type something else and then backspace it to get back to the full list.

The same thing happens with animal selection.

**Comment by mjs** [comment]
Is it possible to add a little space between the selected items? They current end up right up against each other.

image

**Comment by mjs** [comment]
I've just been using the changes in this PR for a while and they're great. This is a much easier way to find what you want.

File: views/getRecordings.pug [hunk 1]
```diff
@@ -34,11 +36,41 @@ html
...
+              .dropdown-item bird/kiwi
+              .dropdown-item dog
+              .dropdown-item hare
+              .dropdown-item human
+              .dropdown-item insect
+              .dropdown-item mouse
+              .dropdown-item rabbit
+              .dropdown-item spider
+              .dropdown-item other
+              .dropdown-divider
```
Comment (mjs) [inline views/getRecordings.pug:65]: nice idea to put the dividers in

**Review by mjs — APPROVED** [review]
The code LGTM. Aside from the small issues mentioned already, I'm happy for this to land.

Thanks so much.

**Comment by arthurmcgregor** [comment]
Sweet thanks @mjs 
I've fixed those few issues you mentioned (hopefully) so will merge this now :-)

**Pull request merged**